src/main/java/org/wso2/carbon/connector/connection/EmailConnection.java

```diff
@@ -0,0 +1,144 @@
+        return props;
+    }
+
+    /**
+     * Sets secure properties
+     *
+     * @param connectionConfiguration configurations to be set
+     * @return Properties to be configured
+     */
+    private Properties setSecureProperties(ConnectionConfiguration connectionConfiguration) {
+
+        Properties props = new Properties();
+        props.setProperty(protocol.getStartTlsProperty(), TRUE);
+        if (connectionConfiguration.isRequireTLS()) {
+            props.setProperty(protocol.getStartTlsProperty(), TRUE);
+        } else {
+            props.setProperty(protocol.getSslEnableProperty(), TRUE);
+            props.setProperty(protocol.getSocketFactoryFallbackProperty(),
+                    EmailConstants.DEFAULT_SOCKETFACTORY_FALLBACK);
+            props.setProperty(protocol.getSocketFactoryPortProperty(),
+                    String.valueOf(connectionConfiguration.getPort()));
+        }
+
+        if (connectionConfiguration.getCipherSuites() != null) {
+            props.setProperty(protocol.getSslCipherSuitesProperty(),
+                    replaceWithWhitespace(connectionConfiguration.getCipherSuites()));
+        }
+
+        if (connectionConfiguration.getSslProtocols() != null) {
+            props.setProperty(protocol.getSslProtocolsProperty(),
+                    replaceWithWhitespace(connectionConfiguration.getSslProtocols()));
+        }
+
+        if (connectionConfiguration.getTrustedHosts() != null) {
+            props.setProperty(protocol.getSslTrustProperty(),
+                    replaceWithWhitespace(connectionConfiguration.getTrustedHosts()));
+        }
+
+        if (connectionConfiguration.isCheckServerIdentity()) {
+            props.setProperty(protocol.getCheckServerIdentityProperty(), Boolean.toString(connectionConfiguration
+                    .isCheckServerIdentity()));
+        }
+
+        return props;
+    }
+
+    /**
+     * Sets timeout properties
+     *
+     * @param readTimeout       Read Timeout
+     * @param writeTimeout      Write Timeout
+     * @param connectionTimeout Connection Timeout
+     * @return timeout properties
+     */
+    private Properties setTimeouts(String readTimeout, String writeTimeout, String connectionTimeout) {
+
+        Properties props = new Properties();
+        if (readTimeout != null) {
+            props.setProperty(protocol.getReadTimeoutProperty(), readTimeout);
+        }
+        if (writeTimeout != null) {
+            props.setProperty(protocol.getWriteTimeoutProperty(), writeTimeout);
+        }
+        if (connectionTimeout != null) {
+            props.setProperty(protocol.getConnectionTimeoutProperty(), connectionTimeout);
+        }
+        return props;
+    }
+
+    /**
+     * Replace a comma in a comma separated string with whitespace
+     *
+     * @param configString Comma separated string
+     * @return Whitespace separated string
+     */
+    private String replaceWithWhitespace(String configString) {
```

Comment:
Why do we need this functionality?

Reply:
JavaMail API expects a whitespace-separated string (https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html). However, since this is not a common practice, IMO it is better we let the user configure a comma-separated string on our end and then later convert it to whitespaces. WDYT?

Reply:
Yes - +1 for the approach.